src/mslice/models/cut/cut_algorithm.py

```diff
@@ -126,27 +126,30 @@ def _compute_cut_nonPSD(selected_workspace, cut_axis, integration_axis, emode, a
     return ws_out
 
 
-def _cut_nonPSD_theta(ax1_binning, ax2_binning, selected_workspace, algo):
-
-    converted_nonpsd = ConvertSpectrumAxis(InputWorkspace=selected_workspace, Target='theta', StoreInADS=False)
-
+def _cut_nonPSD_general(ax1_binning, ax2_binning, selected_workspace, algo):
     if 'Integration' in algo:
         ax1 = [float(x1) for x1 in ax1_binning.split(',')]
         ax2 = [float(x2) for x2 in ax2_binning.split(',')]
         if np.abs((ax1[2]-ax1[0]) - ax1[1]) < 0.0001:
-            ws_out = Integration(InputWorkspace=converted_nonpsd, RangeLower=ax1[0], RangeUpper=ax1[2], EnableLogging=False)
+            ws_out = Integration(InputWorkspace=selected_workspace, RangeLower=ax1[0], RangeUpper=ax1[2], EnableLogging=False)
             ws_out = Transpose(InputWorkspace=ws_out, EnableLogging=False)
             ws_out = Rebin(InputWorkspace=ws_out, Params=ax2_binning, EnableLogging=False)
         else:
-            ws_out = Rebin(InputWorkspace=converted_nonpsd, Params=ax1_binning, EnableLogging=False)
+            ws_out = Rebin(InputWorkspace=selected_workspace, Params=ax1_binning, EnableLogging=False)
             ws_out = Transpose(InputWorkspace=ws_out, EnableLogging=False)
             ws_out = Integration(InputWorkspace=ws_out, RangeLower=ax2[0], RangeUpper=ax2[2], EnableLogging=False)
     else:
-        ws_out = Rebin2D(InputWorkspace=converted_nonpsd, Axis1Binning=ax1_binning, Axis2Binning=ax2_binning,
+        ws_out = Rebin2D(InputWorkspace=selected_workspace, Axis1Binning=ax1_binning, Axis2Binning=ax2_binning,
```

Comment:
When testing on a slice taken from a PSD workspace, this alg threw an error.

There appears to be two issues:
1) We cannot use the algorithm whilst the workspace does not have a name. The name is stripped by `CloneWorkspace` whilst performing a binary op during GDOS correction as the workspace is not stored in the ADS. We can fix this temporarily by cloning the workspace here, storing it in the ADS and giving it a name.
2) Following the above fix a histogram workspace is output rather than a rebinned workspace, which can not be used with `Rebin2D`.

I now remember that this issue was the reason I was summing the slice bins in the first place, rather than using `Rebin` or `cut`.

Do you have any idea on a solution for this? If there is nothing simple, we may have to disable `GDOS` correction on slices taken from `PSD` workspaces for this release.

Reply:
Right, sorry I didn't check this but the original routine with `np.nansum` should work for these workspaces (because they don't have the extra hidden-from-python weights), so I'll restore that bit of the code for PSD workspaces.